### ISSUE TITLE
Eureka registration relies on serialization order of fields inside dataCenterInfo

### ISSUE THREAD

**Issue by jolynch** [Netflix/eureka#1310]
While attempting to register into Eureka with a sidecar agent written in Python I came across a curious behavior where Eureka will fail to register if the JSON serialization places the `name` field of `dataCenterInfo` object after the `metadata` field but it will succeed if placed before. Concretely the following is considered valid:
```json
"dataCenterInfo": {
  "@class": "com.netflix.appinfo.AmazonInfo",
  "name": "Amazon",
  "metadata": {
    "instance-type": "m5.xlarge"
  }
}
```
and the following is rejected as invalid JSON 
```
"dataCenterInfo": {
  "@class": "com.netflix.appinfo.AmazonInfo",
  "metadata": {
    "instance-type": "m5.xlarge"
  },
  "name": "Amazon"
}
```
In response to a payload containing the second `dataCenterInfo` object the client receives a 400 code with `{"error": "cannot parse request body"}` even though both are valid json payloads.

I have a minimal [reproducible test case](https://gist.github.com/jolynch/f862b2d673acaa484da1bf63910da1fd) as well as a unit test [on my fork](https://github.com/jolynch/eureka/commit/f99930d51f782de977cd47828431b8ad6326ecac) that demonstrates the issue.

I imagine that the jackson parsers are somewhat ossified at this point, but I figured documenting the bug may be useful for future users to avoid the sharp edge.

**Comment by brharrington** [comment]
FWIW, there is an existing [test case](https://github.com/Netflix/eureka/blob/master/eureka-client/src/test/java/com/netflix/discovery/converters/jackson/builder/StringInterningAmazonInfoBuilderTest.java#L108) for having name after the metadata. See also #1224.

**Comment by jolynch** [comment]
@brharrington are you saying this is already fixed? Also I'm totally cool with "close won't fix" I just figured it was good to get this into the tracker in case we do want to fix it ... (or so it can help someone else googling the problem).

**Comment by brharrington** [comment]
It should be fixed if you are using [1.9.13](https://github.com/Netflix/eureka/releases/tag/v1.9.13) or newer.

**Comment by jolynch** [comment]
Ah nice, closing then, thanks @brharrington!
